### PR TITLE
Integrate `swift_version` DSL support into pod targets

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/7134

screen_shot_2017-12-01_at_3_12_01_pm

screen_shot_2017-12-01_at_3_14_37_pm

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ gem 'json', :git => 'https://github.com/segiddins/json.git', :branch => 'seg-1.7
 
 group :development do
   cp_gem 'claide',                'CLAide'
-  cp_gem 'cocoapods-core',        'Core'
+  cp_gem 'cocoapods-core',        'cocoapods-core-mine', path: '../cocoapods-core-mine'
```
Comment (dnkoutso) [inline Gemfile]: revert

File: Gemfile.lock [hunk 2]
```diff
@@ -5,16 +5,6 @@ GIT
   specs:
     claide (1.0.2)
 
-GIT
```
Comment (dnkoutso) [inline Gemfile.lock]: revert

**Comment by dnkoutso** [comment]
Should go green.

**Comment by dnkoutso** [comment]
oops one more thing left for validator.

**Comment by dnkoutso** [comment]
All set.

File: lib/cocoapods/validator.rb [hunk 3]
```diff
@@ -395,19 +395,37 @@ def validate_documentation_url(spec)
...
+      unless spec_swift_version.nil?
+        message = nil
+        if !dot_swift_version.nil? && dot_swift_version != spec_swift_version.to_s
+          message = "Specification `#{spec.name}` specifies an inconsistent `swift_version` (`#{spec_swift_version}`) compared to the one present in your `.swift-version` file (`#{dot_swift_version}`). " \
+                    'Please remove the `.swift-version` file which is now deprecated and only use the `swift_version` attribute within your podspec.'
+        elsif !@swift_version.nil? && @swift_version != spec_swift_version.to_s
+          message = "Specification `#{spec.name}` specifies an inconsistent `swift_version` (`#{spec_swift_version}`) compared to the one passed during lint (`#{@swift_version}`)."
+        end
+        unless message.nil?
+          error('swift', message)
```
Comment (justinseanmartin) [inline lib/cocoapods/validator.rb]: NIT: I think it would be cleaner to just inline this personally, but don't feel strongly about that.

**Comment by dnkoutso** [comment]
Found one more edge case on validator. Will fix and update.

File: lib/cocoapods/validator.rb [hunk 4]
```diff
@@ -471,12 +489,17 @@ def install_pod
          perform_post_install_actions).each { |m| @installer.send(m) }
 
       deployment_target = spec.subspec_by_name(subspec_name).deployment_target(consumer.platform_name)
-      @installer.aggregate_targets.each do |target|
+      configure_pod_targets(@installer.aggregate_targets, deployment_target)
+      @installer.pods_project.save
+    end
+
+    def configure_pod_targets(targets, deployment_target)
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb]: for posterity, extracted method so I can test it easier.

**Pull request merged**